XMonad/Layout/LayoutModifier.hs

```diff
@@ -188,11 +188,27 @@ class (Show (m a), Read (m a)) => LayoutModifier m a where
     redoLayout :: m a               -- ^ the layout modifier
                -> Rectangle         -- ^ screen rectangle
                -> Maybe (Stack a)   -- ^ current window stack
-               -> [(a, Rectangle)]  -- ^ (window,rectangle) pairs returned
+               -> [(a, Rectangle)]  -- ^ (window, rectangle) pairs returned
                                     -- by the underlying layout
                -> X ([(a, Rectangle)], Maybe (m a))
     redoLayout m r ms wrs = do hook m; return $ pureModifier m r ms wrs
 
+    -- | 'redoLayoutWithWorkspace' is exactly like 'redoLayout', execept
+    -- that the original workspace is also provided as an argument
+    redoLayoutWithWorkspace :: m a
+                            -- ^ the layout modifier
+                            -> Workspace WorkspaceId (ModifiedLayout m l a) a
+                            -- ^ The original workspace that is being laid out
+                            -> Rectangle
+                            -- ^ screen rectangle
+                            -> Maybe (Stack a)
+                            -- ^ current window stack
+                            -> [(a, Rectangle)]
+                            -- ^ (window, rectangle) pairs returned by the
+                            -- underlying layout
+                            -> X ([(a, Rectangle)], Maybe (m a))
+    redoLayoutWithWorkspace m _ = redoLayout m
```

Comment:
I worry that some may find this unpalatable for some reason, but I believe that the way I have implemented this leads to no change in the current behavior of anything.